winrt/lib/xaml/BaseControl.h

```diff
@@ -502,9 +506,19 @@ namespace ABI { namespace Microsoft { namespace Graphics { namespace Canvas { na
 
         bool IsVisible() const
         {
+            if (m_isVisibleOverrideEnabled)
+            {
+                return(m_isVisibleOverrideValue);
+            }
             return m_isVisible;
         }
 
+        void SetVisibleOverride(bool isVisible)
+        {
+            m_isVisibleOverrideEnabled = true;
+            m_isVisibleOverrideValue = isVisible;
+        }
```

Comment:
Does this need a lock? I have no idea which state needs one and which not. Just looking at the other code.
Maybe also IsVisible() needs one because a possible race.